components/server/src/bitbucket-server/bitbucket-server-repository-provider.ts

```diff
@@ -147,13 +147,15 @@ export class BitbucketServerRepositoryProvider implements RepositoryProvider {
     async getUserRepos(user: User): Promise<string[]> {
         try {
             const repos = await this.api.getRepos(user, { limit: 1000, permission: "REPO_READ" });
+            const result: string[] = [];
+            (repos.values || []).forEach((r) => {
+                const cloneUrl = r.links.clone.find((u) => u.name === "http")?.href;
+                if (cloneUrl) {
+                    result.push(cloneUrl.replace("http://", "https://"));
```

Comment:
I wonder how the test above ever can succeed if we are replacing all http:// to https:// here.

Reply:
Good catch. 
It couldn't have, if I'd re-run them after applying a "last minute" fix for https://linear.app/gitpod/issue/EXP-411#comment-4f6eee75